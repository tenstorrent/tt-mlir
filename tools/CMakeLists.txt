add_subdirectory(ttmlir-opt)
add_subdirectory(ttmlir-lsp-server)
add_subdirectory(ttmlir-translate)
add_subdirectory(explorer)
