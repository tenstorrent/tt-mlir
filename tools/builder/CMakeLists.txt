include(AddMLIRPython)

declare_mlir_python_sources(BuilderSources
  ROOT_DIR "${CMAKE_CURRENT_SOURCE_DIR}"
  SOURCES
    # builder
    base/__init__.py
    base/builder.py
    base/builder_golden.py
    base/builder_utils.py

    # ttir subpackage files
    ttir/__init__.py
    ttir/ttir_builder.py

    # stablehlo subpackage files
    stablehlo/__init__.py
    stablehlo/stablehlo_builder.py

    # d2m subpackage files
    d2m/__init__.py
    d2m/d2m_builder.py

    # ttnn subpackage files
    ttnn/__init__.py
    ttnn/ttnn_builder.py
)

add_mlir_python_modules(BuilderPythonModules
  ROOT_PREFIX "${TTMLIR_PYTHON_PACKAGES_DIR}/builder"
  INSTALL_PREFIX "python_packages/builder"
  DECLARED_SOURCES BuilderSources
)
