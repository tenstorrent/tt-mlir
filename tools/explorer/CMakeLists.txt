include(ExternalProject)

set(TT_EXPLORER_SCRIPT ${CMAKE_CURRENT_SOURCE_DIR}/run.py)
set(TTMLIR_BUILD_BIN_DIR ${TTMLIR_BINARY_DIR}/bin)

set(MODEL_EXPLORER_VERSION "ca884d5eb3291507e7f4e76776957e231b2d9b6d")
ExternalProject_Add(
  model-explorer
  PREFIX ${CMAKE_CURRENT_SOURCE_DIR}/model-explorer
  GIT_REPOSITORY https://github.com/tenstorrent/model-explorer.git
  GIT_TAG ${MODEL_EXPLORER_VERSION}
  GIT_PROGRESS ON
  CONFIGURE_COMMAND ""
  BUILD_COMMAND ""
  INSTALL_COMMAND ""
)

add_custom_target(explorer
  COMMENT "Building tt-explorer... ${TTMLIR_BIN_DIR}"
  COMMAND pip install $<$<CONFIG:Debug>:-e> ${CMAKE_CURRENT_SOURCE_DIR}/tt_adapter
  COMMAND pip uninstall -y ai-edge-model-explorer
  COMMAND ${CMAKE_COMMAND} -E remove_directory ${CMAKE_CURRENT_SOURCE_DIR}/model-explorer/src/model-explorer/src/server/package/build
  COMMAND pip install --upgrade ${CMAKE_CURRENT_SOURCE_DIR}/model-explorer/src/model-explorer/src/server/package

  DEPENDS TTMLIRPythonModules model-explorer ttrt ttmlir-opt ttmlir-translate
)

add_custom_command(TARGET explorer POST_BUILD
  COMMENT "Installing tt-explorer command..."
  COMMAND ${CMAKE_COMMAND} -E create_symlink ${TT_EXPLORER_SCRIPT} ${TTMLIR_BUILD_BIN_DIR}/tt-explorer
  COMMAND ${CMAKE_COMMAND} -E echo "Done. Run: tt-explorer to start the server."
)
