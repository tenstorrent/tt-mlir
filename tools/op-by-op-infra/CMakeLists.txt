include(AddMLIRPython)

declare_mlir_python_sources(TTMLIRPythonOpByOpInfra)

declare_mlir_python_sources(TTMLIRPythonOpByOpInfraSources
  ROOT_DIR "${CMAKE_CURRENT_SOURCE_DIR}/op_by_op_infra"
  ADD_TO_PARENT TTMLIRPythonOpByOpInfra
  SOURCES
    __init__.py
    execution_result.py
    mlir_module_executor.py
    mlir_module_splitter.py
    pydantic_models.py
    utils.py
    workflow_internal.py
    workflow.py
)

add_mlir_python_modules(TTMLIRPythonOpByOpInfraModules
    ROOT_PREFIX "${TTMLIR_PYTHON_PACKAGES_DIR}/op_by_op_infra"
    INSTALL_PREFIX "python_packages/op_by_op_infra"
    DECLARED_SOURCES TTMLIRPythonOpByOpInfraSources
)
