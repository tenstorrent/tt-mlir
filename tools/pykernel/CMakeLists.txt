include(AddMLIRPython)

set(TTPYKERNEL_ROOT_DIR "${CMAKE_CURRENT_SOURCE_DIR}")

declare_mlir_python_sources(TTPykernelSources
    ROOT_DIR "${TTPYKERNEL_ROOT_DIR}"
    SOURCES
        ast.py
        op.py
        types.py
)

add_mlir_python_modules(TTPykernelModules
    ROOT_PREFIX "${TTMLIR_PYTHON_PACKAGES_DIR}/pykernel"
    INSTALL_PREFIX "python_packages/pykernel"
    DECLARED_SOURCES TTPykernelSources
)
