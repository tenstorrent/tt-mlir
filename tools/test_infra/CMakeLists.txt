include(AddMLIRPython)

declare_mlir_python_sources(TTMLIRPythonTestInfra)

declare_mlir_python_sources(TTMLIRPythonTestInfraSources
  ROOT_DIR "${CMAKE_CURRENT_SOURCE_DIR}"
  ADD_TO_PARENT TTMLIRPythonTestInfra
  SOURCES
    ccl_golden.py
    test_optimizer_overrides.py
    test_utils.py
    ttir_builder.py
)

add_mlir_python_modules(TTMLIRPythonTestInfraModules
    ROOT_PREFIX "${TTMLIR_PYTHON_PACKAGES_DIR}/test_infra"
    INSTALL_PREFIX "python_packages/test_infra"
    DECLARED_SOURCES TTMLIRPythonTestInfraSources
)
