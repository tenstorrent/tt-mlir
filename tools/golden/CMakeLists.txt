include(AddMLIRPython)

declare_mlir_python_sources(GoldenSources
  ROOT_DIR "${CMAKE_CURRENT_SOURCE_DIR}"
  SOURCES
    __init__.py
    mapping.py
)

add_mlir_python_modules(GoldenPythonModules
  ROOT_PREFIX "${TTMLIR_PYTHON_PACKAGES_DIR}/golden"
  INSTALL_PREFIX "python_packages/golden"
  DECLARED_SOURCES GoldenSources
)
